flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -393,7 +395,7 @@ class TableImpl(
 
   override def window(overWindows: OverWindow*): OverWindowedTable = {
 
-    if (tableEnv.isInstanceOf[BatchTableEnvironment]) {
+    if (tableEnv.isInstanceOf[BatchTableEnvImpl]) {
```

Comment:
What about the java version?

I think it applies to all places we check if it is Batch or Stream.

Reply:
What do you mean? The `BatchTableEnvImpl` here is `api.BatchTableEnvImpl`. Both `scala.BatchTableEnvImpl` and `java.BatchTableEnvImpl` extend it.

Reply:
Sorry, you are right, ignore my comment.